File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        stringstream ss(s);
        vector<string> v;
        string word;
        while(ss >> word) {
            v.push_back(word);
        }
        string ans = "";
        int n = v.size();
        for(int i=n-1; i>=0; i--) {
            ans += v[i] + " ";
        }
        ans.pop_back();
        return ans;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String x[] = s.trim().split("\\s+");
        String ans = "";
        int n = x.length;
        for(int i=n-1; i>0; i--) {
            ans += x[i] + " ";
        }
        ans += x[0];
        return ans;
    }
}